python/dgl/transform.py

```diff
@@ -1367,4 +1367,125 @@ def as_immutable_graph(hg):
     g.edata.update(hg.edata)
     return g
 
```

Comment:
Not necessarily need to complete in this PR, but I would recommend:
1. Adding an `is_block` flag in DGLHeteroGraph
2. Adding an API like `block_to_graph` for converting a block into a graph.
2. Each transform function should check if the input graph is a block. If this is the case, an warning should be raised for pointing to `block_to_graph`.